### ISSUE TITLE
(Bug) Step4: Transaction is generated after each page refreshing

### ISSUE THREAD

**Issue by dennis00010011b** [poanetwork/token-wizard#1113]
Steps:

 1. Start creating any crowdsale  and fill out the forms with valid parameters until you reach Step 4. Use any network.
2. Refresh the page in Step 4 and confirm popup.
3. Observe count of transaction in Nifty wallet

Expected result:
 - number of transaction should remains the same 

Actual result:
 - number of transaction increased with each refresh 

If you are reporting a problem with Token Wizard, please include the following information:

### Which network did you use? (Mainnet, Kovan, Rinkeby, etc.)

*any*

### If you were able to create it, what is the URL of your crowdsale?

*n/a*

### Do you have screenshots showing the problem?

<img width="1440" alt="44465497-9b2e0600-a5d2-11e8-9ae0-9501917a73be" src="https://user-images.githubusercontent.com/14067096/44682039-21918000-a9f7-11e8-8287-2f2c3d8ac37e.png">

### Do you see errors in the dev console? If yes, please include a screenshot

No

---

**Comment by fernandomg** [comment]
I need confirmation for the approach that I'll be taking on this issue, as at some point it will require the user interaction.

Each step of the deployment process has a list of actions, to make it the most grained possible this can be divided into 4 different actions:
1. Step Waiting to be Processed
1. Transaction Sent and Pending of being signed (this is the user did not confirmed the tx in the wallet yet)
1. Transaction Pending of being mined (txHash generated in this step)
1. Transaction Mined

If I treat each one of those actions as a _flag_, TW will require user interaction for the step `.2`. In this case, the TW can prompt the user to cancel the current Tx pending of approval in the wallet and hit a _Continue_ button, which will trigger the step to be re-sent.

For the step `.3` there'll TW will track the pending transaction by its `transactionHash`, and will continue without the user noticing it. Maybe, in this step, we can prompt the user with a message or highlight the step for the tx pending of being mined.

@dennis00010011b @vbaranov thoghts?

**Comment by vbaranov** [comment]
I agree, that `2.` step cannot be resolved without interaction with a user. I support prompting in this case.

As for the statuses of tx life cycle, maybe we could show 1-3 statuses to a user by means of labels for every current pending tx, like this (I do not pretend to exact styles, but something like this):

![2018-09-20 13 26 18](https://user-images.githubusercontent.com/4341812/45812920-e6bef700-bcd9-11e8-8261-5ccf6179f8b6.png)
![2018-09-20 13 27 40](https://user-images.githubusercontent.com/4341812/45812922-e888ba80-bcd9-11e8-927f-0fa4518f60f0.png)
![2018-09-20 13 34 03](https://user-images.githubusercontent.com/4341812/45812929-eaeb1480-bcd9-11e8-921b-39cde388c97f.png)

Thereby, a user will see what is the current status of pending tx. So, in case of refreshing at `3.` step he/she will see again `tx is pending of being mined...` (without prompting) and after inclusion tx to the block, we move to the next tx. @fernandomg @dennis00010011b - it is open for discussion.
